fedbiomed/common/secagg/_dh.py

```diff
@@ -0,0 +1,159 @@
+from cryptography.hazmat.primitives.asymmetric import ec
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography.hazmat.primitives.kdf.concatkdf import ConcatKDFHash
+from cryptography.hazmat.backends import default_backend
+from fedbiomed.common.exceptions import FedbiomedSecaggCrypterError
+from fedbiomed.common.constants import ErrorNumbers
+
+
+class DHKey:
+    """
+    Key handling for ephemeral-ephemeral ECDH, a session is defined for each experiment.
+
+    This class handles the generation and importing of ECC keys using the P-256 curve. It provides methods
+    to export the private and public keys in PEM format as bytes.
+
+    Attributes:
+        private_key: The user's private ECC key.
+        public_key: The user's public ECC key.
+    """
+
+    def __init__(self, private_key_pem=None, public_key_pem=None):
+        """
+        Initializes the DHKey instance by generating a new key pair or importing the provided keys.
+
+        Args:
+            private_key_pem: Optional. The private key in PEM format.
+            public_key_pem: Optional. The public key in PEM format.
+        """
+        if private_key_pem:
+            try:
+                self._private_key = serialization.load_pem_private_key(
+                    private_key_pem, password=None, backend=default_backend()
+                )
+            except ValueError:
+                raise FedbiomedSecaggCrypterError(
+                    f"{ErrorNumbers.FB628.value}: Invalid private key format"
+                )
+        else:
+            self._private_key = ec.generate_private_key(
+                ec.SECP256R1(), default_backend()
+            )
+
+        self._public_key = self._private_key.public_key()
+
+        if public_key_pem:
+            try:
+                self._public_key = serialization.load_pem_public_key(
+                    public_key_pem, backend=default_backend()
+                )
+            except ValueError:
+                raise FedbiomedSecaggCrypterError(
+                    f"{ErrorNumbers.FB629.value}: Invalid public key format"
+                )
```

Comment:
Public key and private key attributes can be public  so `DHKeyAggreement can access`

 ```suggestion
        if private_key_pem:
            try:
                self.private_key = serialization.load_pem_private_key(
                    private_key_pem, password=None, backend=default_backend()
                )
            except ValueError:
                raise FedbiomedSecaggCrypterError(
                    f"{ErrorNumbers.FB628.value}: Invalid private key format"
                )
        else:
            self.private_key = ec.generate_private_key(
                ec.SECP256R1(), default_backend()
            )

        self.public_key = self._private_key.public_key()

        if public_key_pem:
            try:
                self.public_key = serialization.load_pem_public_key(
                    public_key_pem, backend=default_backend()
                )
            except ValueError:
                raise FedbiomedSecaggCrypterError(
                    f"{ErrorNumbers.FB629.value}: Invalid public key format"
                )
```